build-system/tasks/helpers.js

```diff
@@ -385,6 +385,48 @@ function finishBundle(srcFilename, destDir, destFilename, options) {
   }
 }
 
+/**
+ * Creates a babel plugin for esbuild for the given caller. Optionally enables
+ * caching to speed up transforms.
+ * @param {string} callerName
+ * @param {boolean} enableCache
+ * @param {function()} postStep
+ * @return {!Object}
```

Comment:
would it be possible to formalize some of these generic object returns into typedefs?

Reply:
Thanks for pointing this out. @rileyajones is working on a PR to update all types in `build-system/`. In order to make it easy to update type info for new code like this, we'll need to incorporate some sort of automated workflow that can ideally work with our current IDE auto-fix-on-save workflow. I'm going to leave this for a follow up effort.